### DISCUSSION TITLE
connect to x.x.x.x port 4000 failed: Connection refused

### DISCUSSION THREAD

**Discussion by buzzhd** [teslamate-org/teslamate#2646] [Q&A] [answered]
Hi guys,
Really appreciated if you could help me out. I have been working on teslamate for quite a while and it all stopped working after upgrade to the latest version becuase i kept getting the error message say connection refused (teslamate is on Phicomm N1 with ip address 10.18.18.49):

*   Trying 10.18.18.49:4000...
* connect to 10.18.18.49 port 4000 failed: Connection refused
* Failed to connect to 10.18.18.49 port 4000: Connection refused
* Closing connection 0
curl: (7) Failed to connect to 10.18.18.49 port 4000: Connection refused

Standard docker-compose.yml was used with no modification, meaning it is on both port 4000 and 3000 and etc. Port 3000 is ok, but not 4000.

Port 4000 listening:
tcp        0      0 0.0.0.0:4000            0.0.0.0:*               LISTEN      149384/docker-proxy 
tcp6       0      0 :::4000                 :::*                    LISTEN      149390/docker-proxy 

Also I had some issues with my Synology NAS running Teslamate and once upgraded to latest version and it stopped working with the same error message shown above. 

I thought it might be the firewall settings on router(s), so I was bold enough to disable FW and tested with no luck at all. 

Efforts have been made to downgrade the previous release like 1.25.1 and again problems were still there.

Thanks!

**Comment by buzzhd** [comment]
Alternatively, I just had Teslamate installed on VPS and tested with the following message:

root@TSL:/tesla# curl -v http://x.x.x.x:4000
*   Trying x.x.x.x:4000...
* connect to x.x.x.x port 4000 failed: Connection refused
* Failed to connect to x.x.x.x port 4000: Connection refused
* Closing connection 0
curl: (7) Failed to connect to x.x.x. port 4000: Connection refused

No ports were listening on either 4000 or  3000.

Docker ps shows:

CONTAINER ID   IMAGE                        COMMAND                  CREATED          STATUS                          PORTS                                      NAMES
70b0d5a74242   teslamate/teslamate:latest   "tini -- /bin/sh /en…"   3 minutes ago    Restarting (1) 52 seconds ago                                              tesla_teslamate_1
c72905312b25   eclipse-mosquitto:2          "/docker-entrypoint.…"   3 minutes ago    Up 3 minutes                    1883/tcp                                   tesla_mosquitto_1
b81f7a423518   postgres:13                  "docker-entrypoint.s…"   3 minutes ago    Up 3 minutes                    5432/tcp                                   tesla_database_1
ab3ce264a3c5   teslamate/grafana:latest     "/run.sh"                3 minutes ago    Up 3 minutes                    0.0.0.0:3000->3000/tcp                     tesla_grafana_1
92c6979a84fa   traefik:v2.4                 "/entrypoint.sh --gl…"   23 minutes ago   Up 23 minutes                   0.0.0.0:80->80/tcp, 0.0.0.0:443->443/tcp   tesla_proxy_1

teslamate showing Restarting, no database? Can't even connect to port 4000 so no login can be made, ouch.

Not sure if this is the software bugs or anything and sincerely wish this could be solved at any time soon.

Thanks!

**Comment by buzzhd** [comment]
so may have to wait for the upcoming release to get it fixed and untill then have to live with no teslamate for a while...

**Comment by buzzhd** [comment, marked as answer]
Finally got it working by using the old version of yml file. So definitely it was the software issue here.

**Reply by liuguangyong93** [reply]
please give me one

**Comment by Docfxit** [comment]
I have the exact same issue.  I installed Teslamate on a Synology, I can't get to 192.168.xxx.xxx:4000
Unable to connect
Could I get a copy of your yml file that fixed the issue?
